src/transforms/aws_ec2_metadata.rs

```diff
@@ -170,8 +172,21 @@ impl TransformConfig for Ec2Metadata {
     }
 }
 
-impl FunctionTransform for Ec2MetadataTransform {
-    fn transform(&mut self, output: &mut Vec<Event>, mut event: Event) {
+impl TaskTransform for Ec2MetadataTransform {
```

Comment:
I actually think this would be fine left as a `FunctionTransform`. Cloning it would just clone the `ReadHandle` to the evmap. The state refresh happens in a separate tokio task spawned once at config build time.

Reply:
In this case, it's because evmap is not sync, so we'd have to wrap it in something sync. We'd have to wrap it in something like an `Arc<Mutex<T>>` to do this which would not be great for perf. 😭 

```bash
ana@autonoma:~/git/timberio/vector$ make check-clippy
cargo clippy --workspace --all-targets --features all-integration-tests -- -D warnings
    Checking vector v0.11.0 (/home/ana/git/timberio/vector)
error[E0277]: `std::cell::Cell<()>` cannot be shared between threads safely
   --> src/transforms/aws_ec2_metadata.rs:174:6
    |
174 | impl FunctionTransform for Ec2MetadataTransform {
    |      ^^^^^^^^^^^^^^^^^ `std::cell::Cell<()>` cannot be shared between threads safely
    | 
   ::: src/transforms/mod.rs:152:59
    |
152 | pub trait FunctionTransform: Send + dyn_clone::DynClone + Sync {
    |                                                           ---- required by this bound in `transforms::FunctionTransform`
    |
    = help: within `transforms::aws_ec2_metadata::Ec2MetadataTransform`, the trait `std::marker::Sync` is not implemented for `std::cell::Cell<()>`
    = note: required because it appears within the type `std::marker::PhantomData<std::cell::Cell<()>>`
    = note: required because it appears within the type `evmap::ReadHandle<std::string::String, bytes::Bytes>`
    = note: required because it appears within the type `transforms::aws_ec2_metadata::Ec2MetadataTransform`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `vector`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error[E0277]: `std::cell::Cell<()>` cannot be shared between threads safely
   --> src/transforms/aws_ec2_metadata.rs:174:6
    |
174 | impl FunctionTransform for Ec2MetadataTransform {
    |      ^^^^^^^^^^^^^^^^^ `std::cell::Cell<()>` cannot be shared between threads safely
    | 
   ::: src/transforms/mod.rs:152:59
    |
152 | pub trait FunctionTransform: Send + dyn_clone::DynClone + Sync {
    |                                                           ---- required by this bound in `transforms::FunctionTransform`
    |
    = help: within `transforms::aws_ec2_metadata::Ec2MetadataTransform`, the trait `std::marker::Sync` is not implemented for `std::cell::Cell<()>`
    = note: required because it appears within the type `std::marker::PhantomData<std::cell::Cell<()>>`
    = note: required because it appears within the type `evmap::ReadHandle<std::string::String, bytes::Bytes>`
    = note: required because it appears within the type `transforms::aws_ec2_metadata::Ec2MetadataTransform`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: build failed
make: *** [Makefile:812: check-clippy] Error 101
```

Reply:
Aha, I see, hmm....

It looks like they provide a `ReadHandleryFactory` that is `Sync`: https://docs.rs/evmap/10.0.2/evmap/struct.ReadHandleFactory.html. Generating new `ReadHandle`s from that does require a lock though.

I guess given those limitations of evmap, this seems :+1: